### PR TITLE
zhttp module without tapir dependency; upgrade to stable zio2

### PR DESCRIPTION
**Pull request by soujiro32167**

### CODE REVIEW

File: modules/zio2/testkit/src/main/scala/com/caesars/tracing/TracingTestUtils.scala [hunk 1]
```diff
@@ -104,7 +104,7 @@ object ZTracerImplementationSpecUtils {
 
     for {
       tracedApp1 <- TracedHttp.layer()(app)
-      _          <- tracedApp1(Request(method = Method.GET, url = URL(path = Path(Vector("foo"), trailingSlash = false)))).orDieWith(_.get)
+      _          <- tracedApp1(Request(url = URL.fromString("/foo").getOrElse(???))).either
```
Comment (FrancisToth) [inline modules/zio2/testkit/src/main/scala/com/caesars/tracing/TracingTestUtils.scala:107]: Is this an omission?

**Comment by FrancisToth** [comment]
I'm not sure to understand the title of this PR. Is the goal about upgrading the ZIO 2 module to a stable version and keep tapir to ZIO 1 only?

File: modules/zio2/testkit/src/main/scala/com/caesars/tracing/TracingTestUtils.scala (hunk 1 above)
Comment (soujiro32167) [reply modules/zio2/testkit/src/main/scala/com/caesars/tracing/TracingTestUtils.scala:107]: nope, just simplifying

**Comment by soujiro32167** [comment]
@FrancisToth the goals are:
1. Remove the `tapir` dependency from the `zhttp` modules, since it only depends on zio-http and not tapir. This was needed because we had a binary compatibility issue when importing into `ui-gateway`
2. Upgrade the zio2 modules to the stable `2.0.2` version, instead of the RC version. This is also to avoid RC-based incompatibilities in the future

File: modules/zio2/testkit/src/main/scala/com/caesars/tracing/TracingTestUtils.scala (hunk 1 above)
Comment (FrancisToth) [reply modules/zio2/testkit/src/main/scala/com/caesars/tracing/TracingTestUtils.scala:107]: Then instead, why not storing that in a variable, or even better refactor the tests using a function:
```scala
def mkApp(app: ...) = {
    for {
      tracedApp <- TracedHttp.layer()(app)
      _         <- tracedApp(Request(method = Method.GET, url = URL(path = Path(Vector("foo"), trailingSlash = false)))).either
      ref       <- ZIO.service[SpanRecorder]
      spans     <- ref.get
      span = spans.head
    } yield span
}
```
and then reuse this function everywhere. This simplifies the code and get rid of some boilerplate. Personally, I'm not a big fan of keeping `???` in the code. This sounds like something that is not implemented. Secondly, I'm not sure about what's the type of `URL.fromString`, but it it's an option, then one might better call `get`.

If none of this is possible, then I would simply store the request in a value:
```scala
val get = Request(method = Method.GET, url = URL(path = Path(Vector("foo"), trailingSlash = false)))
```

Comment (soujiro32167) [reply modules/zio2/testkit/src/main/scala/com/caesars/tracing/TracingTestUtils.scala:107]: That is valid and needed, but not what I came into this codebase to do ;)
Lets see if I can improve this...

**Pull request closed without merging**